Sources/MessagingInApp/Views/GistInlineInAppMessageView.swift

```diff
@@ -15,20 +15,255 @@ public protocol GistInlineMessageUIViewDelegate: AnyObject {
+            refreshView()
+        }
+    }
+
+    private var contentSize: CGSize = .zero {
+        didSet {
+            // Notify the system that the intrinsic content size has changed
+            invalidateIntrinsicContentSize()
+        }
+    }
+
+    // How the View communicates the size of the in-app message that's rendered.
+    // UIKit and SwiftUI frameworks use the intrinsicContentSize to adjust auto layout constraints automaticaly to display the View correctly.
+    // Note: If you want to animate when the content size changes, create a wrapper around this View and listen to delegate events. The intrinsicContentSize instantly changes the size.
+    override public var intrinsicContentSize: CGSize {
+        contentSize
+    }
+
+    func updateContentSize(newHeight: CGFloat, newWidth: CGFloat) {
+        contentSize = .init(width: newWidth, height: newHeight)
+    }
+
     public weak var delegate: GistInlineMessageUIViewDelegate?
 
+    // When a fetch request is performed, it's an async operation to have the inline View notified about this fetch and the inline View processing the fetch.
+    // There is currently no easy way to know when the inline View has finished processing the fetch.
+    // This listener is a hack for our automated tests to know when the inline View has finished processing the fetch.
+    //
+    // See linear ticket MBL-427 to learn more about this limitation in our tests.
+    var refreshViewListener: (() -> Void)?
+
+    // Inline messages that have already been shown by this View instance.
+    // This is used to prevent showing the same message multiple times when the close button is pressed.
+    var previouslyShownMessages: [Message] = []
+
+    var inlineMessageManager: InlineMessageManager?
+
+    // Determines if the View is already trying to show a message or not.
+    private var isRenderingOrDisplayingAMessage: Bool {
+        inlineMessageManager != nil
+    }
+
+    // Subscriber to listen to changes in InAppMessageState.
+    private var inAppMessageStoreSubscriber: InAppMessageStoreSubscriber?
+
     public init(elementId: String) {
         super.init(frame: .zero)
         self.elementId = elementId
+
+        // Setup the View and display a message, if one available. Since an elementId has been set.
+        setupView()
+        refreshView()
     }
 
     // This is called when the View is created from a Storyboard.
     required init?(coder: NSCoder) {
         super.init(coder: coder)
+
+        setupView()
+        // An element id will not be set yet. No need to check for messages to display.
+    }
+
+    deinit {
+        unsubscribeFromInAppMessageState()
+    }
+
+    private func setupView() {
+        subscribeToInAppMessageState()
+    }
+
+    private func subscribeToInAppMessageState() {
+        // Begin listening to the queue for new messages.
+        inAppMessageStoreSubscriber = {
+            let subscriber = InAppMessageStoreSubscriber { [weak self] state in
```

Comment:
Updated to subscribe to `InAppMessageState` [instead of `EventBus`](https://github.com/customerio/customerio-ios/blob/111d9d915820fc19c96ecbc802b21823b5613f3e/Sources/MessagingInApp/Views/GistInlineInAppMessageView.swift#L109-L115). Also added `weak self` and unsubscribed in `deinit` as I believe customer app holding a strong reference to view should suffice here. But final testing will be done after sample app updates.